src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -140,12 +105,109 @@ orderby monikerGroup
             return (errors, model, fileManifests);
         }
 
+        private Lazy<(FilePath, PublishItem, ConflictingType, ConcurrentDictionary<FilePath, IReadOnlyList<string>>)> Add(FilePath file, PublishItem item, Action? writeOutput)
+        {
+            if (!_config.DryRun)
+            {
+                writeOutput?.Invoke();
+            }
+            return new Lazy<(FilePath, PublishItem, ConflictingType, ConcurrentDictionary<FilePath, IReadOnlyList<string>>)>(()
+            => (file, item, ConflictingType.None, new ConcurrentDictionary<FilePath, IReadOnlyList<string>>()));
+        }
+
+        private Lazy<(FilePath, PublishItem, ConflictingType, ConcurrentDictionary<FilePath, IReadOnlyList<string>>)> Update(FilePath file, PublishItem item, Action? writeOutput, Lazy<(FilePath, PublishItem, ConflictingType, ConcurrentDictionary<FilePath, IReadOnlyList<string>>)> existingValue, PublishItemKey newKey)
+        {
+            var (existingFile, existingPublishItem, conflictingTYpe, conflictingFiles) = existingValue.Value;
+            conflictingFiles.TryAdd(existingFile, existingPublishItem.Monikers);
+            conflictingFiles.TryAdd(file, item.Monikers);
+
+            if (PublishItemKey.OutputPathConflicts(newKey.OutputPath, existingPublishItem.Path))
```

Comment:
consider let `PublishItem` implement `IComparer<PublishItem>` interface

Reply:
would like to keep `output-path-conflict` and `publish-url-conflict`